doc/4-instrument/fm-esfm.md

```diff
@@ -0,0 +1,119 @@
+
+![FM ADSR chart](FM-ADSRchart.png)
+
+- **Key Scale Rate (KSR)**: also known as "Rate Scale". determines the degree to which the envelope execution speed increases according to the pitch.
+- **Frequency Multiplier (MULT)**: sets the coarse pitch offset in relation to the note (0 to 15). 0 is -1 octave, 1 is 0 octaves, 2 is 1 octave, 3 is 1 octave 7 semitones, and so on.
+  - note that values 11, 13 and 14 behave as 10, 12 and 15 respectively.
+- **Tune (CT)**: sets the semitone offset in relation to the note (-24 to 24).
+  - this is a software effect.
+- **Fine Detune (DT)**: shifts the pitch in fine steps (-128 to 127). 0 is the base pitch, -128 is -1 semitone, 127 is nearly +1 semitone.
+  - this is a software effect.
+- **Left Output (L)** and **Right Output (R)**: toggles output on left/right channels from the operator to the audio output.
+- **Waveform Select (WS)**: changes the waveform of the operator (0 to 7).
+- **Vibrato (VIB)**: makes the operator affected by LFO vibrato.
+- **Vibrato Depth flag (DVB/FMD)**: doubles the effect of LFO vibrato in pitch when set.
+
+### routing controls
+
+- **Output Level (OL)**: sets the output level from this operator to the audio output (0 to 7).
+  - 7 is the loudest level and 1 is the softest, while 0 disables audio output.
+  - this output scaling factor is applied after TL and envelope scaling have been performed.
+  - a change of one unit is about 6 dB.
+- **Modulation Input Level (MI)**: sets the modulation level from the previous operator to this operator (0 to 7).
+  - 7 is the strongest level and 1 is the weakest, while 0 disables modulation.
+  - for operator 1, since there's no previous operator, this controls the **feedback level**.
+  - this modulation scaling factor is applied after the previous operator's TL and envelope scaling have been performed, but is unaffected by OL above.
+  - a change of one unit is about 6 dB.
+
+### table of OL and MI values for obtaining common algorithms
+
+note: MI1 is not included as it is the feedback level.
+
+| algorithm                      | OL1 | MI2 | OL2 | MI3 | OL3 | MI4 | OL4 |
+|--------------------------------|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
+| OPN algorithm **0**            |  0  |  7  |  0  |  7  |  0  |  7  |  7  |
+| OPN algorithm **4**            |  0  |  7  |  7  |  0  |  0  |  7  |  7  |
+| OPN algorithm **6**            |  0  |  7  |  7  |  0  |  7  |  0  |  7  |
+| OPN algorithm **7**            |  7  |  0  |  7  |  0  |  7  |  0  |  7  |
+| OPL3 algorithm **1**           |  7  |  0  |  0  |  7  |  0  |  7  |  7  |
+| OPL3 algorithm **3**           |  7  |  0  |  0  |  7  |  7  |  0  |  7  |
+| OPL3 algorithm **1** (variant) |  0  |  7  |  0  |  7  |  7  |  0  |  7  |
+
+### fixed frequency mode
+
+each operator has a Fixed Frequency mode. once enabled, the operator runs at the specified frequency regardless of the note.
+
+## macros
+
+these macros allow you to control several parameters of FM per tick.
```

Comment:
```suggestion
these macros allow you to control several parameters of FM on each tick.
```

Slightly clearer language (I hope?)

Reply:
This was copypasted from the OPL page, but I totally agree.